### PR TITLE
Feature/update pagination

### PR DESCRIPTION
**Pull request by eskilsen**
Update of pagination script, component and documentation page. 
## Description
New documentation, new style, more userfriendly pagination script.  

## How Has This Been Tested?

Created tests suitable for pagination component.
Render tests for pagination doc page. 
Tests for pagination script. 
Update snapshots. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have updated the **CHANGELOG** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/App/ComponentsDocumentation/components/Pagination/index.js [hunk 1]
```diff
@@ -1,68 +1,109 @@
...
-        </p>
-        <ComponentPreview language="html" showCasePanel codeFigure>
-            <PaginationComponent text="16 of 256" arrows farArrows />
-        </ComponentPreview>
-    </>
+const DeveloperDocumentation = () => (
+    <section>
+        <h2 id="developer-documentation">Developer documentation</h2>
+        <h3>Pagination attributes</h3>
+        <p>Remember to add <CodeTags type="primary" code={"<link rel=\"prev\" href=\"prev_page\">"}/> and <CodeTags type="primary" code={"<link rel=\"next\" href=\"next_page\">"}/> in <CodeTags type="primary" code={"<head>"}/> of your HTML page. These attributes are used to indicate the relations between a sequence of pages to search engines.</p>
```
Comment (asbjornu) [inline src/App/ComponentsDocumentation/components/Pagination/index.js:71]:
```suggestion
        <p>Consider adding <CodeTags type="primary" code={"<link rel=\"…\" href=\"…\">"}/> to the <CodeTags type="primary" code={"<head>"}/> element of your HTML page. These links are used to indicate the relationships between a sequence of pages to web browsers and search engines. Link relations to consider adding are <CodeTags type="primary" code={"first"}/>, <CodeTags type="primary" code={"last"}/>, <CodeTags type="primary" code={"next"}/> and <CodeTags type="primary" code={"prev"}/>.</p>
```

File: src/App/ComponentsDocumentation/components/Pagination/index.js [hunk 2]
```diff
@@ -1,68 +1,109 @@
...
-            If the number of pages get high enough then it is better to use numbers.
-        </p>
-        <ComponentPreview language="html" showCasePanel codeFigure>
-            <PaginationComponent text="16 of 256" arrows farArrows />
-        </ComponentPreview>
-    </>
+const DeveloperDocumentation = () => (
+    <section>
+        <h2 id="developer-documentation">Developer documentation</h2>
+        <h3>Pagination attributes</h3>
```
Comment (asbjornu) [inline src/App/ComponentsDocumentation/components/Pagination/index.js:70]:
```suggestion
        <h3>Link relations</h3>
```

File: src/App/ComponentsDocumentation/components/Pagination/index.js [hunk 3]
```diff
@@ -1,68 +1,109 @@
...
+        <p>Remember to add <CodeTags type="primary" code={"<link rel=\"prev\" href=\"prev_page\">"}/> and <CodeTags type="primary" code={"<link rel=\"next\" href=\"next_page\">"}/> in <CodeTags type="primary" code={"<head>"}/> of your HTML page. These attributes are used to indicate the relations between a sequence of pages to search engines.</p>
+        <h3>Javascript methods</h3>
+        <p>Our pagination component works without a script, so the styling is still working where Javascript is prohibited. But where it is allowed, and where you do not want to spend time banging your head to figure out the algorithm, we have provided a script that paginates the pages for you! <CodeTags type="secondary" code="dg.paginate(pages, currentActive)"/>. The script takes in two props, <CodeTags type="primary" code="pages"/> and <CodeTags type="primary" code="currentActive"/>, where <CodeTags type="primary" code="pages"/> is total pages and <CodeTags type="primary" code="currentActive"/> is the current active page.</p>
```
Comment (asbjornu) [inline src/App/ComponentsDocumentation/components/Pagination/index.js:73]:
```suggestion
        <p>Our pagination component works without a script, so the styling is still working where JavaScript is prohibited. But where it is allowed, and where you do not want to spend time banging your head to figure out the algorithm, the Design Guide provides a reference implementation! <CodeTags type="secondary" code="dg.paginate(pages, currentActive)"/>. The script takes in two arguments; <CodeTags type="primary" code="pages"/> and <CodeTags type="primary" code="currentActive"/>, where <CodeTags type="primary" code="pages"/> is total pages and <CodeTags type="primary" code="currentActive"/> is the current active page.</p>
```

File: src/App/ComponentsDocumentation/components/Pagination/index.js [hunk 4]
```diff
@@ -1,68 +1,109 @@
...
+            console.log(paginate(10, 5)){"\n"}{"\n"}
+
+                {"//expected output:"}{"\n"}
+            [&#123;page: 1, current: false&#125;,{"\n"}
+            &#123;page: null, current: false&#125;,{"\n"}
+            &#123;page: 4, current: false&#125;,{"\n"}
+            &#123;page: 5, current: true&#125;,{"\n"}
+            &#123;page: 6, current: false&#125;,{"\n"}
+            &#123;page: null, current: false&#125;,{"\n"}
+            &#123;page: 10, current: false&#125;]
```
Comment (asbjornu) [inline src/App/ComponentsDocumentation/components/Pagination/index.js:90]:
```suggestion
                {"//expected output:"}{"\n"}
            [
                &#123;page: 1, current: false&#125;,{"\n"}
                &#123;page: null, current: false&#125;,{"\n"}
                &#123;page: 4, current: false&#125;,{"\n"}
                &#123;page: 5, current: true&#125;,{"\n"}
                &#123;page: 6, current: false&#125;,{"\n"}
                &#123;page: null, current: false&#125;,{"\n"}
                &#123;page: 10, current: false&#125;
            ]
```

File: src/App/ComponentsDocumentation/components/Pagination/index.js [hunk 5]
```diff
@@ -1,68 +1,109 @@
...
+            </ComponentPreview>
+        </section>
+    </section>
 );
 
 const Pagination = () => (
     <DocContainer docToc>
         <p className="lead">
-            Use the pagination component to indicate that a series of related content exist across multiple pages.
+            Pagination is used to divide large amounts of content into separate pages and allows the user to navigate to next or previous pages.
```
Comment (asbjornu) [inline src/App/ComponentsDocumentation/components/Pagination/index.js:99]:
```suggestion
            Pagination is used to divide large amounts of content into separate pages and allows the user to navigate to the next or previous pages.
```

File: src/scripts/main/pagination/index.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
+const paginate = (pages, currentActive) => {
+    if (currentActive > pages) {
+        throw "CurrentActive can not be greater then total pages.";
```
Comment (asbjornu) [inline src/scripts/main/pagination/index.js:3]:
```suggestion
        throw "CurrentActive can not be greater than total pages.";
```

File: src/scripts/main/pagination/index.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    if (currentActive <= 0) {
+        throw ("CurrentActive must be greater then 0");
```
Comment (asbjornu) [inline src/scripts/main/pagination/index.js:7]:
```suggestion
        throw ("CurrentActive must be greater than 0");
```

**Review by asbjornu — CHANGES_REQUESTED** [review]
👏🏼

**Review by PhungNg — APPROVED** [review]
Tears of joy!! 😭

File: src/App/ComponentsDocumentation/components/Pagination/index.js [hunk 8]
```diff
@@ -1,68 +1,111 @@
...
+        </div>
+
+        <p>On smaller screens the page links are replaced with text showing the page information, still using arrows to go to next or previous page but with the option to go to the first or last page as well.</p>
+
+        <h3>When to consider something else</h3>
+        <p>For exploration of content, where users are browsing aimlessly for something interesting, infinite scroll is better suited than using pagination. Pagination is best when the user is trying to accomplish a goal, for instance trying to find a particular article from a list.</p>
+
+        <h3>How to use Pagination</h3>
+        <h4>Number of pages</h4>
+        <p>While the pagination component supports an infinite amount of pages, it is difficult to deliver a good user experience when the number of pages exceeds 10. In circumstances where it is difficult for a user to find what they are looking for within the first few search results, they may develop bad habits (such as finding what they are looking for on page 13, search result 5). To mitigate this, more advanced solutions than a single field search should be considered, such as categorization, labels, faceted search, etc</p>
```
Comment (asbjornu) [inline src/App/ComponentsDocumentation/components/Pagination/index.js:39]: I think this reads a bit better.

```suggestion
        <p>While the pagination component supports an infinite amount of pages, it is difficult to deliver a good user experience when the number of pages exceeds 10. In circumstances where it is difficult for a user to find what they are looking for within the first few search results, they may develop bad habits (such as finding what they are looking for on page 13, search result 5). To mitigate this, more advanced solutions than a single search field should be considered, such as categorization, labels, faceted search, etc.</p>
```

**Review by asbjornu — CHANGES_REQUESTED** [review]
A very minor nit, then this is ready for 🚀❗

**Review by asbjornu — APPROVED** [review]
Awesome!

**Pull request merged**